Structure_Engine/Query/SolidVolume.cs

```diff
@@ -106,6 +104,26 @@ public static double SolidVolume(this PileFoundation pileFoundation)
             return pileFoundation.IsNull() ? 0 : pileFoundation.PileCap.SolidVolume() + pileFoundation.Piles.Select(x => x.SolidVolume()).Sum();
         }
 
+        /***************************************************/
+
+        [Description("Returns a Stem's solid volume based on the area of the element and its top and bottom thickness.")]
+        [Input("stem", "The Stem to get the volume from.")]
+        [Output("volume", "The Stem solid material volume.", typeof(Volume))]
+        public static double SolidVolume(this Stem stem)
+        {
+            return stem.IsNull() ? 0 : Engine.Geometry.Query.IArea(stem.Perimeter) * (stem.ThicknessBottom + stem.ThicknessTop) / 2;
```

Comment:
Think this is only true for a rectangular perimiter (with some other requirements on it as well) as well as some paralellogram shapes etc. For, for example, a perimiter looking something like: 
![image](https://github.com/user-attachments/assets/76ab3101-7d86-4848-a423-9a93cbbad61f)
The volume would be incorrect.

Could see this fixed in a couple of different ways:
1. Fully correctly compute the volume for any perimiter shape (dont think this is feasible in this PR)
2. Change the class to enforce rectangular outlines, and then state that the bottom/top thickness relates to that rectangles coordinate system. (might be the msot involved change, and also a bit less flexible as I presume there is a reason you allowed for a PolyCurve/ICurve in the first place)
3. Add checks to see if the formula is correct:
   - Is top and bottom thickness equal, if so, the formula is correct
   - Is the perimiter rectangular and aligned with what is assumed to be top and bottom gradient? If so, formula is correct
   - Else, either raise a warning, stating something about that the volume is approximated, and should not be fully trusted, or raise an error and return NaN. 
4. Always raise a warning stating the requirements, and warn about that it sometimes can be invalid. (quickest to do, but also worst UX I think)

Reply:
I have opted for Option 3 and raised an issue here for future development: https://github.com/BHoM/BHoM_Engine/issues/3409